File: Atividades A1/15 Questões/GO/Questoes15/Questao3/main.go
package main

//3. Encapsulamento Implemente uma classe ContaBancaria com atributos saldo, titular e métodos depositar e sacar. Use encapsulamento para proteger o atributo saldo.
import "fmt"

type ContaBancaria struct {
	titular string
	saldo   float64
}

func (c *ContaBancaria) Depositar(valor float64) {
	if valor > 0 {
		c.saldo += valor
	} else {
		fmt.Println("Valor deve ser maior que 0")
	}
}

func (c *ContaBancaria) Sacar(valor float64) {
	if valor > 0 && valor <= c.saldo {
		c.saldo -= valor
	} else if valor > c.saldo {
		fmt.Println("Saldo insuficiente")
	} else {
		fmt.Println("Valor deve ser maior que 0")
	}
}

func (c *ContaBancaria) GetSaldo() float64 {
	return c.saldo
}

func (c ContaBancaria) Exibir() {
	fmt.Printf("Titular: %s, Saldo: %.2f\n", c.titular, c.saldo)
}

func main() {
	conta := ContaBancaria{titular: "Tyler", saldo: 4000}

	conta.Exibir()

	conta.Depositar(700)
	conta.Exibir()

	conta.Sacar(3000)
	conta.Exibir()

	conta.Sacar(5000)
	conta.Exibir()
}

File: Atividades A1/15 Questões/Python/Questoes15/Questao3/Main.py
#3. Encapsulamento Implemente uma classe ContaBancaria com atributos saldo, titular e métodos depositar e sacar. Use encapsulamento para proteger o atributo saldo.

class ContaBancaria:
    def __init__(self, titular, saldo):
        self.titular = titular
        self.saldo = saldo

    def depositar(self, valor):
        if valor > 0:
            self.saldo += valor
        else:
            print('Valor deve ser maior que 0')

    def sacar(self, valor):
        if valor > 0 and valor <= self.saldo:
            self.saldo -= valor
        elif valor > self.saldo:
            print('Saldo insuficiente')
        else:
            print('Valor deve ser maior que 0')

    def getsaldo(self):
        return self.saldo

    def exibir(self):
        print(f'Titular: {self.titular}, Saldo: {self.saldo}')


if __name__ == '__main__':
    conta = ContaBancaria('Tyler', 4000)

    conta.exibir()

    conta.depositar(700)
    conta.exibir()
    conta.sacar(3000)
    conta.exibir()
    conta.sacar(5000)
    conta.exibir()
